src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+
+        if(diagnoser_container_log != ""):
+            dns_check, storage_space_available = check_cluster_DNS(diagnoser_container_log, filepath_with_timestamp, storage_space_available)
+            outbound_connectivity_check, storage_space_available = check_cluster_outbound_connectivity(diagnoser_container_log, filepath_with_timestamp, storage_space_available)
+        else:
+            return "Incomplete", storage_space_available
+
+        if(dns_check == "Passed" and outbound_connectivity_check == "Passed"):
+            return "Passed", storage_space_available
+        elif(dns_check == "Incomplete" or outbound_connectivity_check == "Incomplete"):
+            return "Incomplete", storage_space_available
+        else:
+            return "Failed", storage_space_available
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to perform diagnoser continer check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Diagnoser_Container_Check_Fault_Type, summary="Error occured while performing the diagnoser container checks")
+        cli_outputs.append("An expection has occured while trying to perform diagnoser continer check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Incomplete", storage_space_available
+
+
+def executing_diagnoser_job(corev1_api_instance, batchv1_api_instance, namespace, absolute_path):
+
+    global cli_outputs
+
+    # Setting the log output as Empty
+    diagnoser_container_log = ""
+    yaml_file_path = os.path.join(absolute_path, "troubleshoot_diagnoser_job.yaml")
+    cmd_delete_job = ["kubectl", "delete", "-f", ""]
+    cmd_delete_job[3] = str(yaml_file_path)
+
+    # To handle the user keyboard Interrupt
+    try:
+
+        # Executing the diagnoser_job.yaml
+        config.load_kube_config()
+        k8s_client = client.ApiClient()
+
+        try:
+            utils.create_from_yaml(k8s_client, yaml_file_path)
+        except Exception as e:
+            all_exceptions_list = str(e).split('\n')
+            counter = 0
+            for ind_exception in all_exceptions_list:
+                if("Error from server (Conflict)" in ind_exception or ind_exception == ""):
+                    continue
+                else:
+                    if(counter == 0):
+                        cli_outputs.append("An Error occured while trying to execute diagnoser_job.yaml.")
+                        logger.warning("An Error occured while trying to execute diagnoser_job.yaml.")
+                        counter = 1
+                        logger.warning(ind_exception)
+                    else:
+                        logger.warning(ind_exception)
+            print()
+            if(counter == 1):
+                subprocess.run(cmd_delete_job, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
```

Comment:
we can use Popen here directly instead of subprocess.run. Example https://github.com/Azure/azure-cli-extensions/blob/e28e889aa0d367d9b034be96208154f0fdd4082d/src/connectedk8s/azext_connectedk8s/custom.py#L893

Reply:
Made the changes.